src/components/Search/SearchRouter/SearchRouterInput.tsx

```diff
@@ -0,0 +1,41 @@
+import React, {useState} from 'react';
+import BaseTextInput from '@components/TextInput/BaseTextInput';
+import useThemeStyles from '@hooks/useThemeStyles';
+import variables from '@styles/variables';
+import CONST from '@src/CONST';
+
+type SearchRouterInputProps = {
+    isFullWidth: boolean;
+    onChange: (searchTerm: string) => void;
+    onSubmit: () => void;
+};
+
+function SearchRouterInput({isFullWidth, onChange, onSubmit}: SearchRouterInputProps) {
+    const styles = useThemeStyles();
+
+    const [value, setValue] = useState('');
+
+    const onChangeText = (text: string) => {
+        setValue(text);
+        onChange(text);
+    };
+
+    const modalWidth = isFullWidth ? styles.w100 : {width: variables.popoverWidth};
+
+    return (
+        <BaseTextInput
+            value={value}
+            onChangeText={onChangeText}
+            onSubmitEditing={onSubmit}
+            autoFocus
```

Comment:
Coming from this Issue https://github.com/Expensify/App/issues/51010, We are adding autoFocus here, but it doesn’t always work consistently on Android. To handle this, we’ve an existing prop (shouldDelayFocus) for TextInput that enables autoFocus with a delay specifically for Android https://github.com/Expensify/App/issues/51010#issuecomment-2419429623